xonsh/xoreutils/yes.py

```diff
@@ -0,0 +1,25 @@
+import os
+import sys
+
+
+def yes(args, stdin, stdout, stderr):
+    if '--help' in args:
```

Comment:
Compare with GNU's `yes`:

```
/bin/yes --help | head  # Shows help
/bin/yes blah --help | head  # Normal yes behaviour
/bin/yes --blah --help | head  # Raises error
/bin/yes -- blah blah | head  # Normal yes behaviour (but removes "--")
/bin/yes -- --help | head  # Normal yes behaviour (but removes "--")
/bin/yes -- --blah --help | head  # Normal yes behaviour (but removes "--")
/bin/yes -- -- -- | head  # Normal yes behaviour (final string printed is "-- --")
```
Essentially:

* Switches are parsed only up until the first non-switch
* `--` marks the end of switches (only if they haven't ended already)
* Invalid switches raise an error.

These behaviours are incompatible with an approach relying on `in`, which will always be brittle and will result in the builtin `yes` being incapable of printing some strings.